iceoryx_posh/include/iceoryx_posh/internal/roudi/shared_memory_manager.hpp

```diff
@@ -192,6 +192,10 @@ class SharedMemoryManager
     void deletePortsOfProcess(std::string processName);
     void deleteRunnableAndItsPorts(std::string runnableName);
 
+    void destroySenderPort(SenderPortType::MemberType_t* const senderPortData);
```

Comment:
If we add destroy to the base port we could add here a single method called ```destroyPort``` which takes the basePortData pointer

Reply:
I guess not because there are different list for different ports and different actions have to be taken

Reply:
You are right ... but in the future the ports will maybe be stored in the typed mempool and are essentially shared_ptr, then we could take a look into this again.

Reply:
Yes, but then shared pointers will be use and destroy will be a shared pointer going finally out of scope